client/platform/desktop/backend/serializers/nist.ts

```diff
@@ -63,7 +63,7 @@ interface NistActivity {
 }
 
 function confirmNistFormat(data: NistFile) {
-  return data.activities !== undefined && data.filesProcessed !== undefined;
+  return Array.isArray(data.activities) && Array.isArray(data.filesProcessed);
```

Comment:
Either the interface is incorrect or this will never return false because those files can only be an array.

https://github.com/Kitware/dive/pull/883/files#diff-9179ddbd5a97d67ddf6d93b78f1534efec3cb8f44e3070606a5c0aad890666bcR38-R49

Reply:
https://github.com/Kitware/dive/pull/883/files#diff-9179ddbd5a97d67ddf6d93b78f1534efec3cb8f44e3070606a5c0aad890666bcR76

https://github.com/Kitware/dive/pull/883/files#diff-9179ddbd5a97d67ddf6d93b78f1534efec3cb8f44e3070606a5c0aad890666bcR271

I'm doing this backwards in that I should restructure the confirmNist to handle `any` or some sort of JSON object, confirm that it is a NISTFile and then rely on that to be the Type guard for the `as NistFile` assignment

Reply:
Oh, I see.  Please use a type guard instead.   The parameter to `confirmNistFormat` should be `any` if you don't know what it is going in. 

```
function isNistFile(data: any): data is NistFile {
  return Array.isArray(data.activities) && Array.isArray(data.filesProcessed);
}
```

then, inside a block where `isNistFile(data)` is true, typescript knows `data` is of type `NistFile`